aiidalab_widgets_base/databases.py

```diff
@@ -204,179 +204,234 @@ def _update_structure(self, change: dict) -> None:
+            style=self.STYLE,
         )
-        self.inp_domain.observe(self._domain_changed, names=["value", "options"])
+        self.domain_selector.observe(self._domain_changed, names=["value", "options"])
 
         # Select computer.
-        self.inp_computer = ipw.Dropdown(
-            options=[],
+        self.computer_selector = ipw.Dropdown(
+            options=(),
             description="Computer:",
             disabled=False,
+            layout=self.LAYOUT,
+            style=self.STYLE,
+        )
+        self.computer_selector.observe(
+            self._computer_changed, names=["value", "options"]
         )
-        self.inp_computer.observe(self._computer_changed, names=["value", "options"])
 
         # Select code.
-        self.inp_code = ipw.Dropdown(
-            options=[],
+        self.code_selector = ipw.Dropdown(
+            options=(),
             description="Code:",
             disabled=False,
+            layout=self.LAYOUT,
+            style=self.STYLE,
         )
-        self.inp_code.observe(self._code_changed, names=["value", "options"])
+        self.code_selector.observe(self._code_changed, names=["value", "options"])
 
-        btn_reset = ipw.Button(description="Reset")
-        btn_reset.on_click(self._reset)
+        reset_button = ipw.Button(description="Reset")
+        reset_button.on_click(self.reset)
+
+        if show_reset_button is False:
+            reset_button.layout.display = "none"
 
         super().__init__(
             children=[
-                self.inp_domain,
-                self.inp_computer,
-                self.inp_code,
-                btn_reset,
+                self.domain_selector,
+                self.computer_selector,
+                self.code_selector,
+                reset_button,
             ],
             **kwargs,
         )
-        self.update()
-        self._reset()
-
-    def _reset(self, _=None):
-        self.inp_domain.value = None
-
-    def clean_up_database(self, database, plugin):
-        for domain in list(database.keys()):
-            for computer in list(database[domain].keys() - {"default"}):
-                for code in list(
-                    database[domain][computer].keys()
-                    - {"computer-configure", "computer-setup"}
-                ):
-                    if (
-                        plugin
-                        != database[domain][computer][code]["default_calc_job_plugin"]
+        self.database_source = database_source
+        self.reset()
+
+    def reset(self, _=None):
+        """Reset widget and traits"""
+        with self.hold_trait_notifications():
+            self.domain_selector.value = None
+            self.computer_selector.value = None
+            self.code_selector.value = None
+
+    @tl.observe("database_source")
+    def _database_source_changed(self, _=None):
+        self.database = self._database_generator(
+            self.database_source, self.default_calc_job_plugin
```

Comment:
I was going to make `_database_generator` a `staticmethod` but since it calls the `self._calc_job_plugin_match` so the `self` is needed. 
For the `database_source` and `default_calc_job_plugin`, I think you are right, it is not changed after the object was instantiated. 
I changed it.